ui/packages/catalog-ui-search/src/main/webapp/component/location-old/location-serialization.js

```diff
@@ -170,72 +207,100 @@ const BoundingBox = {
 }
 
 const MultiPolygon = {
-  'json->location': ({
-    geometry: { coordinates },
-    properties: { buffer } = {},
-  }) => ({
-    mode: 'multipolygon',
-    polygon: coordinates,
-    polygonBufferWidth,
-    polygonBufferUnits,
-    polyType: 'multipolygon',
-  }),
-  'location->json': ({
-    polygon = [],
-    polygonBufferWidth,
-    polygonBufferUnits,
-  }) => ({
-    type: 'Feature',
-    geometry: {
-      type: 'MultiPolygon',
-      coordinates: is3DArray(polygon) ? [polygon] : polygon,
-    },
-    properties: {
-      type: 'MultiPolygon',
-      buffer: {
-        width: polygonBufferWidth,
-        unit: polygonBufferUnits,
+  'json->location': json => {
+    const {
+      geometry: { coordinates },
+      properties: { buffer } = {},
+    } = json
+
+    const { width = 0, unit = 'meters' } = buffer
+
+    const polygon = coordinates.map(([child]) => child)
+
+    return {
+      mode: 'poly',
```

Comment:
This will fail with the serializer, since we rely on mode existing for `multipolygon`, which we aren't going to use, since we are now using `poly`. Need to proxy logic in Polygon serialization case to check if array is 3-dimensional and if it is, send it to multipolygon and return from there.

Reply:
Attempted fix: https://github.com/codice/ddf/pull/4478/commits/902987b684a1f915bb8b49c61a6ce4bcde0ade6f